app/src/features/projects/view/components/FundingSource.tsx

```diff
@@ -63,56 +64,118 @@ const FundingSource: React.FC<IProjectFundingProps> = (props) => {
     }
   });
 
-  const [fundingFormData, setFundingFormData] = useState<IProjectFundingFormArrayItem>(
-    ProjectFundingFormArrayItemInitialValues
-  );
+  const [fundingFormData, setFundingFormData] = useState({
+    index: 0,
+    values: ProjectFundingFormArrayItemInitialValues
+  });
 
   const showErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
     setErrorDialogProps({ ...errorDialogProps, ...textDialogProps, open: true });
   };
 
   const [openEditDialog, setOpenEditDialog] = useState(false);
 
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+
   const handleDialogEditOpen = async (itemIndex: number) => {
-    const fundingSource = funding.fundingSources[itemIndex];
+    let fundingSourceValues: IProjectFundingFormArrayItem;
 
-    setFundingFormData({
-      id: fundingSource.id,
-      agency_id: fundingSource.agency_id,
-      investment_action_category: fundingSource.investment_action_category,
-      investment_action_category_name: fundingSource.investment_action_category_name,
-      agency_project_id: fundingSource.agency_project_id,
-      funding_amount: fundingSource.funding_amount,
-      start_date: getFormattedDate(DATE_FORMAT.ShortDateFormat, fundingSource.start_date),
-      end_date: getFormattedDate(DATE_FORMAT.ShortDateFormat, fundingSource.end_date),
-      revision_count: fundingSource.revision_count
-    });
+    if (itemIndex < funding.fundingSources.length) {
```

Comment:
maybe you are already working on it but we need to add tests for the changes in this file :)

Reply:
I want to talk to you about this one ... can we get the PR through for the client to test? and I will implement this shortly.

Reply:
Yeah lets not wait too long to get this merged, if the only remaining bits are test related.  Need to give Robin time to put her stamp of approval on it.  We can always tweak the tests while its in her hands.